`aws-sdk-cpp` build failure on x86_64-linux as of `3fedfb5f`

## Issue description
Build of `aws-sdk-cpp` failed on x86_64-linux as of 3fedfb5f8ad1755140ba13382aa1e453fccf6e02. This is currently breaking python3Packages.optax.

```
error: builder for '/nix/store/f1k2593w12pp4i3gzvhd40ykl31vds20-aws-sdk-cpp-1.9.150.drv' failed with exit code 2;
       last 10 log lines:
       >       |                               ^~~~~~~~~
       > /build/source/aws-cpp-sdk-cognitoidentity-integration-tests/IdentityPoolOperationTest.cpp: In static member function 'static bool {anonymous}::IdentityPoolOperationTest::WaitForIdentitiesToBeCreated(const String&, const std::shared_ptr<Aws::CognitoIdentity::CognitoIdentityClient>&)':
       > /build/source/aws-cpp-sdk-cognitoidentity-integration-tests/IdentityPoolOperationTest.cpp:111:31: error: 'sleep_for' is not a member of 'std::this_thread'
       >   111 |             std::this_thread::sleep_for(std::chrono::seconds(1));
       >       |                               ^~~~~~~~~
       > [ 89%] Linking CXX shared library libaws-cpp-sdk-s3-encryption.so
       > [ 89%] Built target aws-cpp-sdk-s3-encryption
       > make[2]: *** [aws-cpp-sdk-cognitoidentity-integration-tests/CMakeFiles/aws-cpp-sdk-cognitoidentity-integration-tests.dir/build.make:77: aws-cpp-sdk-cognitoidentity-integration-tests/CMakeFiles/aws-cpp-sdk-cognitoidentity-integration-tests.dir/IdentityPoolOperationTest.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:13760: aws-cpp-sdk-cognitoidentity-integration-tests/CMakeFiles/aws-cpp-sdk-cognitoidentity-integration-tests.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/f1k2593w12pp4i3gzvhd40ykl31vds20-aws-sdk-cpp-1.9.150.drv'.
error: 1 dependencies of derivation '/nix/store/21qf4jikrp858y5ck6c7pkbpxzpfinm5-arrow-cpp-7.0.0.drv' failed to build
building '/nix/store/ch6jgjnqviqvhysdx0dl924v75733qlj-python3.9-dm-tree-unstable-2021-12-20.drv'...
error: 1 dependencies of derivation '/nix/store/4p2a26wz7a2s5pcr6iig788zqqg1cd17-python3.9-pyarrow-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s335pxsjyagqa1wwgbyzvqiiq33flr1i-python3.9-apache-beam-2.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6x5c0yc5r1p372nfcyal3sh090r4j7kl-python3.9-tensorflow-datasets-4.5.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wcx65hgbdxz36s43faaqcrb8m5h3g88r-python3.9-optax-0.1.1.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2186440298)

python3Packages.optax maintainers cc: @ndl
aws-sdk-cpp maintainers cc: @edolstra @orivej
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 3fedfb5f8ad1755140ba13382aa1e453fccf6e02
2. Run `nix-build -A aws-sdk-cpp`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1021-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - nixpkgs: `/nix/store/r902mrj2iwsjhy8gs6y0rbr9nada0z4z-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2186440298) that created this issue.
- Internal tag: -5229024172647481710

--- a/pkgs/development/libraries/aws-sdk-cpp/default.nix
+++ b/pkgs/development/libraries/aws-sdk-cpp/default.nix
@@ -1,7 +1,19 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, cmake, curl, openssl, s2n-tls, zlib
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, curl
+, openssl
+, s2n-tls
+, zlib
 , aws-crt-cpp
-, aws-c-cal, aws-c-common, aws-c-event-stream, aws-c-io, aws-checksums
-, CoreAudio, AudioToolbox
+, aws-c-cal
+, aws-c-common
+, aws-c-event-stream
+, aws-c-io
+, aws-checksums
+, CoreAudio
+, AudioToolbox
 , # Allow building a limited set of APIs, e.g. ["s3" "ec2"].
   apis ? ["*"]
 , # Whether to enable AWS' custom memory management.
@@ -18,16 +30,31 @@ in
 
 stdenv.mkDerivation rec {
   pname = "aws-sdk-cpp";
-  version = "1.9.150";
+  version = "1.9.238";
 
   src = fetchFromGitHub {
     owner = "aws";
     repo = "aws-sdk-cpp";
     rev = version;
-    sha256 = "sha256-fgLdXWQKHaCwulrw9KV3vpQ71DjnQAL4heIRW7Rk7UY=";
+    sha256 = "sha256-pEmsTfZXsvJMV79dGkjDNbUVajwyoYgzE5DCsC53pGY=";
   };
 
   postPatch = ''
+    # Missing includes for GCC11
+    sed '5i#include <thread>' -i \
+      aws-cpp-sdk-cloudfront-integration-tests/CloudfrontOperationTest.cpp \
+      aws-cpp-sdk-cognitoidentity-integration-tests/IdentityPoolOperationTest.cpp \
+      aws-cpp-sdk-dynamodb-integration-tests/TableOperationTest.cpp \
+      aws-cpp-sdk-elasticfilesystem-integration-tests/ElasticFileSystemTest.cpp \
+      aws-cpp-sdk-lambda-integration-tests/FunctionTest.cpp \
+      aws-cpp-sdk-mediastore-data-integration-tests/MediaStoreDataTest.cpp \
+      aws-cpp-sdk-queues/source/sqs/SQSQueue.cpp \
+      aws-cpp-sdk-redshift-integration-tests/RedshiftClientTest.cpp \
+      aws-cpp-sdk-s3-crt-integration-tests/BucketAndObjectOperationTest.cpp \
+      aws-cpp-sdk-s3-integration-tests/BucketAndObjectOperationTest.cpp \
+      aws-cpp-sdk-s3control-integration-tests/S3ControlTest.cpp \
+      aws-cpp-sdk-sqs-integration-tests/QueueOperationTest.cpp \
+      aws-cpp-sdk-transfer-tests/TransferTests.cpp
     # Flaky on Hydra
     rm aws-cpp-sdk-core-tests/aws/auth/AWSCredentialsProviderTest.cpp
     # Includes aws-c-auth private headers, so only works with submodule build
@@ -82,12 +109,6 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./cmake-dirs.patch
-
-    # fix cmake config
-    (fetchpatch {
-      url = "https://github.com/aws/aws-sdk-cpp/commit/b102aaf5693c4165c84b616ab9ffb9edfb705239.diff";
-      sha256 = "sha256-38QBo3MEFpyHPb8jZEURRPkoeu4DqWhVeErJayiHKF0=";
-    })
   ];
 
   # Builds in 2+h with 2 cores, and ~10m with a big-parallel builder.

